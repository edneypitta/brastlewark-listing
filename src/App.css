.App {
  background-color: rgb(24, 23, 67);
  min-height: 100vh;
  color: white;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.App header {
  min-height: 20vh;
  font-size: calc(10px + 2vmin);
}

.App footer {
  margin: 20px 0;
}