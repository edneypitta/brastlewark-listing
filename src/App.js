import React, { Component } from 'react'
import './App.css'
import Modal from './shared/Modal'
import SearchBar from './shared/SearchBar'
import GnomeList from './GnomeList'
import GnomeDetails from './GnomeDetails'

const gnomesUrl = 'https://raw.githubusercontent.com/rrafols/mobile_test/master/data.json'

const sortByName = gnomes => gnomes.sort((a, b) => a.name.localeCompare(b.name))

class App extends Component {
  constructor(props) {
    super(props)
    this.state = {
      modal: { isOpen: false },
      gnomes: []
    }
  }

  handleSearchTextChange = ({ target: { value: searchText } }) =>
    this.setState({ searchText })

  toggleModal = gnome =>
    this.setState({
      modal: {
        isOpen: !this.state.modal.isOpen,
        gnome
      }
    })

  getFilteredGnomes = () =>
    this.state.gnomes
      .filter(
        gnome => !this.state.searchText ||
          gnome.name.toLowerCase().includes(this.state.searchText.toLowerCase()))

  componentDidMount() {
    fetch(gnomesUrl)
      .then(response => response.json())
      .then(data => data.Brastlewark)
      .then(sortByName)
      .then(gnomes => this.setState({ gnomes }))
  }

  render() {
    const { searchText, modal } = this.state

    return (
      <div className="App">
        <header>
          <h1>Brastlewark Gnomes</h1>
        </header>
        <main>
          <SearchBar
            value={searchText}
            onChange={this.handleSearchTextChange} />
          <GnomeList
            gnomes={this.getFilteredGnomes()}
            onClick={this.toggleModal} />
          <Modal
            show={modal.isOpen}
            onClose={this.toggleModal.bind(this, null)}>
            {modal.isOpen &&
              <GnomeDetails gnome={modal.gnome} />}
          </Modal>
        </main>
        <footer>
          Made with â™¥ by Edney Pitta
        </footer>
      </div>
    )
  }
}

export default App
