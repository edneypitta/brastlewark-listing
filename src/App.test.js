import React from 'react'
import { mount, shallow } from 'enzyme'
import App from './App'

describe('App', () => {
  it('renders without crashing', () => {
    const component = mount(<App />)
    expect(component).toMatchSnapshot()
  })

  describe('getFilteredGnomes', () => {
    it('filters gnomes by searchText ignoring case', () => {
      const component = shallow(<App />)
      component.instance().setState({
        gnomes: [{ name: 'gnome 1' }, { name: 'gnome 2' }],
        searchText: 'GNOME 1'
      })

      const gnomes = component.instance().getFilteredGnomes()
      expect(gnomes).toHaveLength(1)
    })
  })

  it('renders GnomeDetails if modal is open', () => {
    const component = shallow(<App />)
    component.instance().setState({
      modal: {
        isOpen: true,
        gnome: {}
      }
    })

    expect(component.find('GnomeDetails')).toHaveLength(1)
  })
})
