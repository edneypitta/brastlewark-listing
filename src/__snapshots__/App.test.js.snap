// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
<App>
  <div
    className="App"
  >
    <header>
      <h1>
        Brastlewark Gnomes
      </h1>
    </header>
    <main>
      <SearchBar
        onChange={[Function]}
        value=""
      >
        <div
          className="wrapper"
        >
          <input
            onChange={[Function]}
            placeholder="Search"
            type="text"
            value=""
          />
        </div>
      </SearchBar>
      <GnomeList
        gnomes={Array []}
        onClick={[Function]}
      >
        <ul
          className="gnome-list"
        />
      </GnomeList>
      <Modal
        onClose={[Function]}
        show={false}
      />
    </main>
    <footer>
      Made with â™¥ by Edney Pitta
    </footer>
  </div>
</App>
`;
